rust/datafusion/src/logicalplan.rs

```diff
@@ -784,6 +784,18 @@ pub enum LogicalPlan {
         /// Whether the CSV file contains a header
         has_header: bool,
     },
+    /// Produces a relation with string representations of
+    /// various parts of the plan
+    Explain {
```

Comment:
In a subsequent PR, I would like to suggest we consider making this `Explain(ExplainNode)` or something, and having a separate `struct ExplainNode` with these fields. There is a bunch of boiler plate code for matching and reconstructing `LogicalPlanner` nodes that I think would be less awkward (and require fewer changes to add / modify fields)  if they were collected into a struct